assets/dev/js/frontend/handlers/base-carousel.js

```diff
@@ -89,11 +99,14 @@ export default class CarouselHandlerBase extends SwiperHandlerBase {
 			};
 		}
 
-		if ( showDots ) {
+		if ( showPagination ) {
 			swiperOptions.pagination = {
-				el: '.swiper-pagination',
-				type: 'bullets',
+				el: `.elementor-element-${ this.getID() } .swiper-pagination`,
+				type: !! elementSettings.pagination ? elementSettings.pagination : 'bullets',
 				clickable: true,
+				renderBullet: ( index, classname ) => {
+					return `<span class="${ classname }" data-bullet-index="${ index }" aria-label="${ elementorFrontend.config.i18n.a11yCarouselPaginationBulletMessage } ${ index + 1 }"></span>`;
+				},
```

Comment:
@hein-obox Why is this needed? We are rendering this via PHP, no?

Reply:
1. The bullets are generated with JS by the SwiperJS library.
2. I need to customise the html a little bit.
3. This is standard function provided by the SwiperJS API.